pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeDataTable.java

```diff
@@ -211,12 +209,12 @@ public void remove() {
         public Pair<byte[], byte[]> next() {
           byte[] dimBuff = new byte[dimensionSizeInBytes];
           byte[] metBuff = new byte[metricSizeInBytes];
-          mappedByteBuffer.copyTo(pointer * totalSizeInBytes, dimBuff, 0, dimensionSizeInBytes);
-//          mappedByteBuffer.position(pointer * totalSizeInBytes);
-//          mappedByteBuffer.get(dimBuff);
+          copyTo(mappedByteBuffer, pointer * (long) totalSizeInBytes, dimBuff, 0, dimensionSizeInBytes);
```

Comment:
mappedByteBuffer.toDirectByteBuffer(pointer * (long) totalSizeInBytes, dimensionSizeInBytes) results in an array, you still need to iterate the array to get the content. Probably you are checking a newer version of the library?